AWSQueueManager.php

```diff
@@ -108,17 +108,31 @@ public function getQueues($refresh=false)
     }
 
     /**
+     * Important Note: If the SQS queue is FIFO type we MUST send the MessageGroupId also,
+     * otherwise the request will fail.
+     * @link https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+     *
+     * Define if the SQS queue is a FIFO one via passing the `isFifo` as true on $options param.
+     *
      * @param string $url     url of the queue to send message
      * @param string $message message to send
      * @param array  $options extra options for the message
      * @return boolean message was succesfull
      */
-    public function send($url, $message, $options=array())
+    public function send($url, $message, $options= [])
     {
-        $this->_sqs->sendMessage(array_merge(array(
+        $isFifo = $options['isFifo'] ?? false;
+
+        $messageData = [
             'QueueUrl'    => $url,
             'MessageBody' => $message,
-        ),$options));
+        ];
+
+        if ($isFifo) {
+            $messageData['MessageGroupId'] = uniqid();
```

Comment:
Since we are allowing a new `isFifo` option, I feel we should allow to set the message group id as a option as well, an only use `uniqid()` when not set.

From my understanding, if every message has a different `MessageGroupId`, then the queue won't ack as FIFO, the messages could be out of order.

I believe, in our case, all our async event messages should have the same `MessageGroupId`. Or we could use 3 values, different for normal, high and KPI events for example.

Reply:
This method sends a single message, for this reason I wanted to initialize the `MessageGroupId ` only if the FIFO queue is the case.
On the sendBatch [method](https://github.com/PeoplePerHour/yii-aws-sqs/pull/4/files#diff-22b48a712215b05cae82175741ed10a4R159) the `MessageGroupId` is the same for the batch.

Reply:
That's not my understanding of how `MessageGroupId` works.

> The tag that specifies that a message belongs to a specific message group. Messages that belong to the same message group are processed in a FIFO manner (however, messages in different message groups might be processed out of order).
([source](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html))

Therefore, for our async events to be processed in order, we'll need to always use the same `MessageGroupId`.

Reply:
I got you point @tomfotherby 👍 
I removed the `isFifo` check from the options param, and if the the `MessageGroupId` exists means that the Queue is FIFO.